arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -1239,6 +1237,38 @@ def rule_fr89(
             )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr91(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR91: If the annual report contains information about both the general meeting date
+    (gsd:DateOfGeneralMeeting) and the annual accounts meeting date (gsd:DateOfApprovalOfReport), the values must be the same.
+    """
+    modelXbrl = val.modelXbrl
+    factsPair = groupFactsByContextHash(modelXbrl.factsByQname.get(pluginData.dateOfGeneralMeetingQn, set()).union(modelXbrl.factsByQname.get(pluginData.dateOfApprovalOfReportQn, set())))
```

Comment:
This is probably overkill as TM30 and TM31 both check that if either of these concepts are tagged, they are only tagged once. So you could probably just simplify this check by pulling the facts for each concept and grabbing the first one in the set if one is returned. TM29 checks to make sure that at least one of the concepts is tagged.